src/reset.c

```diff
@@ -47,12 +47,20 @@ static void _show_reset_label(bool status)
 void reset_reset(bool status)
 {
     keystore_lock();
+    uint8_t retries = 0, max_retries = 5;
```

Comment:
better to simply use `int` (more common and probably more efficient than uint8_t), and `const` for max_retries

this function also does 6 attempts not 5, but it's hard to read because the do..while with in-line increment is kind of hard to read. how about doing a simpler but easier to read loop like this:

```
   ATCA_STATUS result;
    int retries = 0;
    const int max_retries = 5;

    while (retries < max_retries) {
        if (atcab_random(rand_out) == ATCA_SUCCESS) {
            return true;
        }
        retries++;
    }
    return false;
```

Reply:
okay, i was trying to make it faster and shorter. but of course i can be more verbose

Reply:
clarity & correctness first :smile: shorter is not better if it's harder to read, and in this case it had a "bug" (too many retries) as a result